modules/logging/filters.go

```diff
@@ -185,15 +187,113 @@ func (m IPMaskFilter) Filter(in zapcore.Field) zapcore.Field {
 	return in
 }
 
+type ActionType string
+
+const (
+	ReplaceType ActionType = "replace"
+	DeleteType  ActionType = "delete"
+)
+
+type cookieFilterAction struct {
+	Type  ActionType `json:"type"`
+	Name  string     `json:"name"`
+	Value string     `json:"value,omitempty"`
+}
+
+// CookiesFilter is a Caddy log field filter that
+// filters cookies.
+type CookieFilter struct {
+	Actions []cookieFilterAction `json:"actions"`
+}
+
+// CaddyModule returns the Caddy module information.
+func (CookieFilter) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "caddy.logging.encoders.filter.cookie",
+		New: func() caddy.Module { return new(CookieFilter) },
+	}
+}
+
+// UnmarshalCaddyfile sets up the module from Caddyfile tokens.
+func (m *CookieFilter) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	for d.Next() {
+		for d.NextBlock(0) {
+			cfa := cookieFilterAction{}
+			switch d.Val() {
+			case "replace":
+				if !d.NextArg() {
+					return d.ArgErr()
+				}
+
+				cfa.Type = ReplaceType
+				cfa.Name = d.Val()
+
+				if !d.NextArg() {
+					return d.ArgErr()
+				}
+				cfa.Value = d.Val()
+
+			case "delete":
+				if !d.NextArg() {
+					return d.ArgErr()
+				}
+
+				cfa.Type = DeleteType
+				cfa.Name = d.Val()
+
+			default:
+				return d.Errf("unrecognized subdirective %s", d.Val())
+			}
+
+			m.Actions = append(m.Actions, cfa)
+		}
+	}
+	return nil
+}
+
+// Filter filters the input field.
+func (m CookieFilter) Filter(in zapcore.Field) zapcore.Field {
+	originRequest := http.Request{Header: http.Header{"Cookie": []string{in.String}}}
+	cookies := originRequest.Cookies()
+	transformedRequest := http.Request{Header: make(http.Header)}
+
+OUTER:
+	for _, c := range cookies {
+		for _, a := range m.Actions {
+			if c.Name != a.Name {
+				continue
+			}
+
+			switch a.Type {
+			case ReplaceType:
+				c.Value = a.Value
+				transformedRequest.AddCookie(c)
+				continue OUTER
+
+			case DeleteType:
+				continue OUTER
```

Comment:
Might be worth mentioning in the godoc that if two "actions" are configured for the same cookie name, only the first will apply (because the `continue OUTER` will skip the others), although it's pretty obvious only one should be configured per.

Could be useful later to apply more than one, but that'll depend on whether any new actions added would make that beneficial.